modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4IncrementalRequestHandlingIT.java

```diff
@@ -218,6 +223,86 @@ public void testClientBackpressure() throws Exception {
         }
     }
 
+    // ensures that server reply 100-continue on acceptable request size
+    public void test100Continue() throws Exception {
+        try (var ctx = setupClientCtx()) {
+            for (int reqNo = 0; reqNo < randomIntBetween(2, 10); reqNo++) {
+                var id = opaqueId(reqNo);
+                var acceptableContentLength = randomIntBetween(0, maxContentLength());
+
+                // send request header and await 100-continue
+                var req = httpRequest(id, acceptableContentLength);
+                HttpUtil.set100ContinueExpected(req, true);
+                ctx.clientChannel.writeAndFlush(req);
+                var resp = (FullHttpResponse) safePoll(ctx.clientRespQueue);
+                assertEquals(HttpResponseStatus.CONTINUE, resp.status());
+                resp.release();
+
+                // send content
+                var content = randomContent(acceptableContentLength, true);
+                ctx.clientChannel.writeAndFlush(content);
+
+                // consume content and reply 200
+                var handler = ctx.awaitRestChannelAccepted(id);
+                handler.consumeBytes(acceptableContentLength);
+                handler.sendResponse(new RestResponse(RestStatus.OK, ""));
+
+                resp = (FullHttpResponse) safePoll(ctx.clientRespQueue);
+                assertEquals(HttpResponseStatus.OK, resp.status());
+                resp.release();
+            }
+        }
+    }
+
+    // ensures that server reply 413-too-large on oversized request with expect-100-continue
+    public void test413TooLargeOnExpect100Continue() throws Exception {
+        try (var ctx = setupClientCtx()) {
+            for (int reqNo = 0; reqNo < randomIntBetween(2, 10); reqNo++) {
+                var id = opaqueId(reqNo);
+                var oversized = maxContentLength() + 1;
+
+                // send request header and await 413 too large
+                var req = httpRequest(id, oversized);
+                HttpUtil.set100ContinueExpected(req, true);
+                ctx.clientChannel.writeAndFlush(req);
+                var resp = (FullHttpResponse) safePoll(ctx.clientRespQueue);
+                assertEquals(HttpResponseStatus.REQUEST_ENTITY_TOO_LARGE, resp.status());
+                resp.release();
+
+                // terminate request
+                ctx.clientChannel.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+            }
+        }
+    }
+
+    // ensures that server reply 413-too-large on oversized chunked encoding request and closes connection
+    public void test413TooLargeOnChunkedEncoding() throws Exception {
```

Comment:
I don't think this is the right behaviour. We might have already started processing the chunked request, but `413` implies we are completely rejecting it according to [RFC 9110 §15.5.14](https://www.rfc-editor.org/rfc/rfc9110.html#section-15.5.14):

> The 413 (Content Too Large) status code indicates that the server is refusing to process a request because the request content is larger than the server is willing or able to process.

Instead, we need to let the `RestHandler` deal with this situation, so that different handlers can respond in different ways. For instance the `_bulk` API should report the usual doc-level responses for any successfully-processed docs, and should return a doc-level `429` for docs past the limit. Other APIs that aggregate the whole request up-front can still reasonably return a `413`.

Reply:
Now I remember, we talked about it. I tried to make it similar to what http-object-aggregator does. I will remove rejection for chunked request.